services/autoscaling/src/simcore_service_autoscaling/utils/utils_docker.py

```diff
@@ -324,6 +327,82 @@ async def get_docker_swarm_join_bash_command() -> str:
     )
 
 
+def get_docker_login_on_start_bash_command(registry_settings: RegistrySettings) -> str:
+    return " ".join(
+        [
+            "docker",
+            "login",
+            "--username",
+            registry_settings.REGISTRY_USER,
+            "--password",
+            registry_settings.REGISTRY_PW.get_secret_value(),
+            registry_settings.resolved_registry_url,
+        ]
+    )
+
+
+_DOCKER_COMPOSE_CMD: Final[str] = "docker compose"
+_PRE_PULL_COMPOSE_PATH: Final[Path] = Path("/docker-pull.compose.yml")
+_DOCKER_COMPOSE_PULL_SCRIPT_PATH: Final[Path] = Path("/docker-pull-script.sh")
+_CRONJOB_LOGS_PATH: Final[Path] = Path("/var/log/docker-pull-cronjob.log")
+
+
+def get_docker_pull_images_on_start_bash_command(
+    docker_tags: list[DockerGenericTag],
+) -> str:
+    if not docker_tags:
+        return ""
```

Comment:
Question: can this actually occur, that you have no images to pull? I'd guess this will be the case when an older project is opened?

Reply:
yes that is the case when the function is called where there are no tags given. so there is nothing to pull.